sormas-ui/src/main/java/de/symeda/sormas/ui/docgeneration/QuarantineOrderLayout.java

```diff
@@ -0,0 +1,132 @@
+import com.vaadin.ui.themes.ValoTheme;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.ReferenceDto;
+import de.symeda.sormas.api.docgeneneration.QuarantineOrderFacade;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.utils.ValidationException;
+
+public class QuarantineOrderLayout extends VerticalLayout {
+
+	private final ReferenceDto caseReferenceDto;
+
+	private final Button createButton;
+	private final VerticalLayout additionalVariablesComponent;
+	private FileDownloader fileDownloader;
+
+	public QuarantineOrderLayout(ReferenceDto referenceDto) {
+		super();
+		this.caseReferenceDto = referenceDto;
+
+		additionalVariablesComponent = new VerticalLayout();
+		additionalVariablesComponent.setSpacing(false);
+		additionalVariablesComponent.setVisible(false);
+
+		createButton = new Button(I18nProperties.getCaption(Captions.actionCreate));
+		createButton.addStyleName(ValoTheme.BUTTON_PRIMARY);
+		createButton.setIcon(VaadinIcons.FILE_TEXT);
+		createButton.setEnabled(false);
+
+		ComboBox<String> templateSelector = new ComboBox<>(I18nProperties.getCaption(Captions.DocumentTemplate_QuarantineOrder));
+		templateSelector.setWidth(80F, Unit.PERCENTAGE);
+		templateSelector.setItems(FacadeProvider.getQuarantineOrderFacade().getAvailableTemplates());
+		templateSelector.addValueChangeListener(e -> {
+			String templateFile = e.getValue();
+			boolean isValidTemplateFile = templateFile != null && !templateFile.isEmpty();
+			createButton.setEnabled(isValidTemplateFile);
+			additionalVariablesComponent.removeAllComponents();
+			additionalVariablesComponent.setVisible(false);
+			if (isValidTemplateFile) {
+				try {
+					List<String> additionalVariables = FacadeProvider.getQuarantineOrderFacade().getAdditionalVariables(templateFile);
+					if (additionalVariables != null && !additionalVariables.isEmpty()) {
+						for (String variable : additionalVariables) {
+							TextField variableInput = new TextField(variable);
+							variableInput.setWidth(80F, Unit.PERCENTAGE);
+							additionalVariablesComponent.addComponent(variableInput);
+						}
+						additionalVariablesComponent.setVisible(true);
+					}
+					setStreamResource(templateFile);
+				} catch (ValidationException validationException) {
+					validationException.printStackTrace();
+					new Notification("Error I18N", "Error Description I18N", Notification.Type.ERROR_MESSAGE, false).show(Page.getCurrent());
+				}
+			}
+		});
+
+		addComponent(templateSelector);
+		addComponent(additionalVariablesComponent);
+		addComponent(createButton);
+	}
+
+	private Properties readAdditionalVariables() {
+		Properties properties = new Properties();
+		forAllVariableInputs(textField -> {
+			properties.setProperty(textField.getCaption(), textField.getValue());
+			return null;
+		});
+		return properties;
+	}
+
+	private void forAllVariableInputs(Function<TextField, Void> function) {
```

Comment:
Unclear method name; I think renaming it to "doForAllVariableInputs" should do the trick

Reply:
Method renamed.